File: src/main/java/net/slipp/support/jpa/UpdatedDateEntityListener.java
package net.slipp.support.jpa;

import java.util.Date;

import javax.persistence.PrePersist;
import javax.persistence.PreUpdate;

public class UpdatedDateEntityListener {

	@PreUpdate
	@PrePersist
	public void prePersistAndUpdate(HasUpdatedDate hud) {
		hud.setUpdatedDate(new Date());
	}
}

File: src/main/scala/net/slipp/support/jpa/UpdatedDateEntityListener.scala
package net.slipp.support.jpa

import java.util.Date
import javax.persistence.PrePersist
import javax.persistence.PreUpdate

class UpdatedDateEntityListener {
  @PreUpdate
  @PrePersist
  def prePersistAndUpdate(hud: HasUpdatedDate) {
    hud.setUpdatedDate(new Date)
  }
}
